### PR TITLE
Issue #39 Update FCLTU for V4 and V5

### LINKED ISSUE #39: Update FCLTU to for differences between v4 and v5
[v4 (912.1-B-3)](http://mtc-m16c.sid.inpe.br/col/sid.inpe.br/mtc-m18@80/2009/07.22.13.02/doc/ccsds%20912.1-B-3.pdf)

[v5 (912.1-B-4)](https://public.ccsds.org/Pubs/912x1b4.pdf)

### PR DESCRIPTION
**Pull request by lorsposto**
Resolves #39

- Updates FCLTU PDUs for the differences between v4 and v5
- Adds version negotiation handling to CLTU to patch `CltuUserToProviderPdu` and `CltuUserToProviderPdu` with v4.
- Adds cltu throw-event test to test patching

GetParameter is not implemented yet and so there is no GetParameter test.

### CODE REVIEW

File: ait/dsn/sle/cltu.py [hunk 1]
```diff
@@ -324,6 +324,23 @@ def decode(self, message):
...
+            if module is None:
+                raise ImportError('No module named %s' % modname)
+            replcls = getattr(module, clsname)
+            if replcls is None:
+                raise ImportError('No class named %s.%s' % (modname, clsname))
+            ait.core.log.info("Replacing %s with class %s" % (clsname, replcls))
+            globals()[clsname] = replcls
+
+        replace('CltuUserToProviderPdu')
+        replace('CltuUserToProviderPdu')
```
Comment (MJJoyce) [inline ait/dsn/sle/cltu.py]: Is this duplicate call intentional or was this meant to be a UserToProvider and ProviderToUser swap?

**Comment by MJJoyce** [comment]
I'm looking at this again and I'm really not liking the idea of replacing classes like this. I suppose we could avoid this mess if we require that users configure what version of the interfaces they want to use so we can intelligently import stuff. That's not really friendly if we're allowing version negotiation with the harness but I think that's functionality that isn't really needed. Projects are going to have negotiated and set in stone interface details so there isn't going to be a need to negotiate down a version.

I realize it means more changes to this but what do you think about that @lorsposto? Have a config option for picking interface versions and it defaults to v5 if nothing is specified. At least then we can intelligently import modules without having to fiddle around with overwrites and whatnot.

**Comment by lorsposto** [comment]
@MJJoyce Agreed; updated the version logic to check ait.config before importing either v4 or v5.

File: ait/dsn/sle/cltu.py [hunk 2]
```diff
@@ -23,13 +23,16 @@
 '''
 import binascii
 import struct
+import pydoc
```
Comment (MJJoyce) [inline ait/dsn/sle/cltu.py]: Are we using this important for anything?

File: ait/dsn/sle/common.py [hunk 3]
```diff
@@ -94,6 +94,7 @@ class SLE(object):
     _handlers = defaultdict(list)
     _data_queue = gevent.queue.Queue()
     _invoke_id = 0
+    _supported_versions = [4, 5]
```
Comment (MJJoyce) [inline ait/dsn/sle/common.py]: Can drop this after the import changes I think.

**Pull request merged**